src/ert/_c_wrappers/enkf/res_config.py

```diff
@@ -528,6 +528,12 @@ def site_config_file(self):
     def ert_templates(self):
         return self._templates
 
+    def get_num_cpu(self) -> Optional[int]:
+        queue_num_cpu = self.queue_config.num_cpu
+        if queue_num_cpu == 0:
```

Comment:
Should we check for against 0 or negative numbers?

Reply:
good question - let me see. there are two paths to queue config (that i consider), config content and config dict - config dict is easy to see (in `queue_config.py`), if the key is not in the dict, the default value is 0, cool.

same goes for the struct of queue config on the C side (`queue_config.cpp`), which gets alloced / inited in `queue_config_alloc_empty`, which gets called by alloc default, which gets called by alloc, which gets called with the config content. all good.

we're really just checking for the default value here, actually. but that is hard to figure out by looking at the code. and now i remember that Eivind suggested adding a flag, something like `num_cpu_set`, in queue config - that would make the code easier to read. let me take a at that again!

Reply:
I saw the queue_config being checked first, then the ecl_config being used if the former was zero. I was just wondering what would happen if any of those returned a negative int.

Reply:
ok test which gave a negative number in a config dict and just inited an enkf main object worked without complaints. so i took it one step further, and ran a local example (`poly_example`) with setting `NUM_CPU -18` in the `poly.ert` config file, and then ran
```
ert ensemble_smoother poly.ert
```

it ran fine, without complaints or problems. adding a quick print statement confirmed that the # of cpus, according to enkf main, was -18, and that's what gets passed to other parts of the code that actually use that number.

of course this is using the `local` queue driver, it might have other consequences using other queue drivers.